integration/test/basic/basic_test.go

```diff
@@ -63,25 +59,57 @@ func TestChartConfigChartsInstalled(t *testing.T) {
 		return
 	}
 
-	guestNamespace := "giantswarm"
 	guestG8sClient := g.G8sClient()
 
 	// Wait for chart configs as they may not have been created yet.
 	err := waitForChartConfigs(guestG8sClient)
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
+}
+
+func TestChartConfigPatchDeploy(t *testing.T) {
+	// These versions have no chartconfigs so we return early.
+	clusterOperatorVersion := os.Getenv("CLOP_VERSION_BUNDLE_VERSION")
+	if clusterOperatorVersion == "0.1.0" || clusterOperatorVersion == "0.2.0" {
+		return
+	}
+
+	guestG8sClient := g.G8sClient()
 
 	chartConfigList, err := guestG8sClient.CoreV1alpha1().ChartConfigs(guestNamespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
-
-	// At least 1 chartconfig is expected. Actual number depends on how many
-	// components have been migrated.
+	// At least 1 chartconfig is required.
 	if len(chartConfigList.Items) == 0 {
 		t.Fatalf("expected at least 1 chartconfigs: %d found", len(chartConfigList.Items))
 	}
+	chartConfigName := chartConfigList.Items[0].Spec.Chart.Name
+
+	patch := ChartConfigDeployPatch{
+		Spec{
+			Chart: Chart{
```

Comment:
This looks like mixing anonymous and defined properties. Lets please not mix them. I prefer the key/value definition anyways.

Reply:
Try to align it with the strucuture in e2etest, also removed the anonymous fields. But not 100% thats what you meant. Don't really know what you mean with key/value definition in the context of nested structs too. Maybe you can enlighten me @xh3b4sd ;)

Reply:
You sorted it already apparently. 

```
		Spec{
			Chart: Chart{
```

```
		Spec: Spec{
			Chart: Chart{
```